dandi/upload.py

```diff
@@ -108,6 +101,14 @@ def upload(
 
     lgr.debug("Working with collection %s", collection_rec)
 
+    try:
+        girder.lookup(client, girder_collection, path=girder_top_folder)
+    except girder.GirderNotFound:
+        raise ValueError(
```

Comment:
please add a quick test for this codepath (upload to some not registered dataset)

Reply:
Done.